File: experiment-5/runs/language=typescript_model=claude-opus-4-7_tooling=beads/rep2/src/dates.ts
// Parse the several date formats found across the CSV sources into ISO YYYY-MM-DD.

export function parseDate(raw: string | null | undefined): string | null {
  if (!raw) return null;
  const s = String(raw).trim();
  if (!s || s === "NA" || s === "-") return null;

  // 2023-09-24 or 2012-05-19 18:30:00 -> 2023-09-24
  const iso = s.match(/^(\d{4})-(\d{2})-(\d{2})/);
  if (iso) return `${iso[1]}-${iso[2]}-${iso[3]}`;

  // 29/03/2003
  const br = s.match(/^(\d{2})\/(\d{2})\/(\d{4})/);
  if (br) return `${br[3]}-${br[2]}-${br[1]}`;

  return null;
}

export function parseSeason(raw: string | number | null | undefined): number | null {
  if (raw === null || raw === undefined) return null;
  const s = String(raw).trim();
  if (!s || s === "NA") return null;
  const n = parseInt(s, 10);
  return Number.isFinite(n) ? n : null;
}

export function parseGoal(raw: string | number | null | undefined): number | null {
  if (raw === null || raw === undefined) return null;
  const s = String(raw).trim();
  if (!s || s === "NA" || s === "-") return null;
  const n = parseFloat(s);
  return Number.isFinite(n) ? Math.round(n) : null;
}

File: experiments/adrianco/experiment-14/runs/language=java_model=opus-4.8-fast_prompt=TDD/rep1/src/main/java/com/example/brsoccer/csv/Dates.java
/*
 * ============================================================================
 * Brazilian Soccer MCP Server
 * ----------------------------------------------------------------------------
 * File:    Dates.java
 * Purpose: Tolerant date parsing for the heterogeneous match files, which mix
 *          ISO dates ("2023-09-24"), ISO date-times ("2012-05-19 18:30:00")
 *          and Brazilian day/month/year dates ("29/03/2003"). Always returns a
 *          LocalDate (time component discarded) or null when unparseable.
 * Part of: csv package (field-level parsing helpers).
 * ============================================================================
 */
package com.example.brsoccer.csv;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

/** Parses the several date formats present across the datasets. */
public final class Dates {

    private static final DateTimeFormatter BR_SLASH = DateTimeFormatter.ofPattern("dd/MM/yyyy");

    private Dates() {
    }

    /** Parse a date string into a {@link LocalDate}, or null if blank/unrecognized. */
    public static LocalDate parse(String raw) {
        if (raw == null) {
            return null;
        }
        String s = raw.trim();
        if (s.isEmpty()) {
            return null;
        }
        // Drop any time component separated by a space (e.g. "2012-05-19 18:30:00").
        int space = s.indexOf(' ');
        String datePart = space > 0 ? s.substring(0, space) : s;
        try {
            if (datePart.contains("/")) {
                return LocalDate.parse(datePart, BR_SLASH);
            }
            return LocalDate.parse(datePart);
        } catch (RuntimeException e) {
            return null;
        }
    }
}
